ingest-app/src/cmr/ingest/services/jobs.clj

```diff
@@ -196,27 +212,38 @@
 (defn- process-subscriptions
   "Process each subscription in subscriptions."
   [context subscriptions time-constraint]
-  (for [subscription subscriptions
-       :let [email-address (get-in subscription [:extra-fields :email-address])
-             coll-id (get-in subscription [:extra-fields :collection-concept-id])
-             query-string (-> (:metadata subscription)
-                              (json/decode)
-                              (get "Query"))
-             query-params (create-query-params query-string)
-             params1 (merge {"created-at" time-constraint}
-                            {"collection-concept-id" coll-id}
-                            query-params)
-             params2 (merge {"revision-date" time-constraint}
-                            {"collection-concept-id" coll-id}
-                            query-params)
-             gran-ref1 (search/find-granule-references context params1)
-             gran-ref2 (search/find-granule-references context params2)
-             gran-ref (distinct (concat gran-ref1 gran-ref2))]]
-      (if (seq gran-ref)
-        ;; These strings are just for debugging purpose. In another ticket
-        ;; I will figure out how to send email.
-        (str "Granules found for: " email-address)
-        (str "No granules found for: " email-address))))
+  (doseq [subscription subscriptions
+         :let [email-address (get-in subscription [:extra-fields :email-address])
+               coll-id (get-in subscription [:extra-fields :collection-concept-id])
+               query-string (-> (:metadata subscription)
+                                (json/decode true)
+                                :Query)
+               query-params (create-query-params query-string)
+               params1 (merge {:created-at time-constraint}
+                              {:collection-concept-id coll-id}
+                              query-params)
+               params2 (merge {:revision-date time-constraint}
+                              {:collection-concept-id coll-id}
+                              query-params)]]
+      (try
+        (let [gran-ref1 (search/find-granule-references context params1)
+              gran-ref2 (search/find-granule-references context params2)
+              gran-ref (distinct (concat gran-ref1 gran-ref2))
+              gran-ref-location (map :location gran-ref)]
+          (when (seq gran-ref)
+            (info "Sending email for subscription: " (:metadata subscription))
```

Comment:
Is there any value in logging the whole subscription metadata here for every send email event? And the sending and Finished sending log messages are asymmetrical. I don't like that. Do you?

Reply:
Also is there any value in logging the `gran-ref-location`?

Reply:
I have responded earlier that I want to do this for debugging purpose at the beginning since I can't do the send-message locally.  Once all looks good, I can log less.

Reply:
Then we should add adequate comments to indicate this, so we can remember to change it later.